salt/pillar/__init__.py

```diff
@@ -322,7 +322,9 @@ def compile_pillar(self, *args, **kwargs):  # Will likely just be pillar_dirs
             else:
                 # We found the minion but not the env. Store it.
                 fresh_pillar = self.fetch_pillar()
-                self.cache[self.minion_id][self.pillarenv] = fresh_pillar
+                self.cache[self.minion_id] = dict(self.cache[self.minion_id], **{
+                    self.pillarenv: fresh_pillar
+                })
```

Comment:
I am not sure what I am looking at. The `self.cache[self.minion_id]` contains a dict, to which `fresh_pillar` variable is assigned. Now, all you do here is just create that dict over, with the same values, ending up with the same result.

What exactly this is helping?

Reply:
@isbm 
Please check https://github.com/saltstack/salt/blob/develop/salt/utils/cache.py#L84

Reply:
That is just a link to the CacheDisk class. Can you please explain better?

Reply:
@rongzeng54 Thanks for giving me a link to a class. Now please explain what you are doing and how this helps.

Reply:
> @rongzeng54 Thanks for giving me a link to a class. Now please explain what you are doing and how this helps.

> self.cache[self.minion_id][self.pillarenv] = x
Will only call the `__getitem__` of CacheDisk class, and will not call the `__setitem__`

> self.cache[self.minion_id] = x
Will only call the `__setitem__` of CacheDisk class

In here, the second is correct